### PR TITLE
Improve write to file

### PR DESCRIPTION
**Pull request by thefourtheye**
`writeToFile` function uses `exists` which is deprecated. This patch replaces that with `access` function.
Apart from that, the `writeToFile` function is fully asynchronous now. This patch removes `sinon` dependency as well. `writeToFile` tests have been written in such a way that the spying is not necessary.

### CODE REVIEW

File: scripts/release-post.js [hunk 1]
```diff
@@ -181,12 +181,21 @@ function renderPost (results) {
 function writeToFile (results) {
   const filepath = path.resolve(__dirname, '..', 'locale', 'en', 'blog', 'release', `v${results.version}.md`)
 
-  if (fs.existsSync(filepath)) {
-    return Promise.reject(new Error(`Release post for ${results.version} already exists!`))
-  }
+  return new Promise(function (resolve, reject) {
+    fs.access(filepath, fs.R_OK | fs.W_OK, function (err) {
+      if (!err) {
```
Comment (lpinca) [inline scripts/release-post.js]: Shouldn't this be `if (err) {`?

Comment (thefourtheye) [inline scripts/release-post.js]: @lpinca If there is no error, it means that the file exists. [`fs.access`](https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback)

Comment (lpinca) [inline scripts/release-post.js]: Correct, my bad.

Comment (lpinca) [inline scripts/release-post.js]: Can't we just use `fs.F_OK`?

``` js
fs.access(filepath, function (err) {
```

File: tests/scripts/release-post.test.js [hunk 2]
```diff
@@ -258,10 +257,11 @@ test('findLatestVersion<version>', (t) => {
 })
 
 test('writeToFile<object>', (t) => {
-  let fileExists
+  let accessResult
+
   const fs = {
-    existsSync: () => fileExists,
-    writeFileSync: sinon.spy()
+    access: (filename, flags, cb) => accessResult ? cb(accessResult) : cb(),
```
Comment (lpinca) [inline tests/scripts/release-post.test.js]: I think that keeping `fileExists` makes it easier to grok.

``` js
access: (filename, cb) => fileExists ? cb() : cb(New Error('ENOENT')),
```

File: scripts/release-post.js (hunk 1 above)
Comment (thefourtheye) [inline scripts/release-post.js]: @lpinca Sure, I'll change it.

File: tests/scripts/release-post.test.js (hunk 2 above)
Comment (thefourtheye) [inline tests/scripts/release-post.test.js]: @lpinca You are right. Replaced it now. PTAL.

**Comment by thefourtheye** [comment]
@lpinca Addressed your comments. PTAL.

**Comment by lpinca** [comment]
LGTM.

File: scripts/release-post.js [hunk 3]
```diff
@@ -181,12 +181,21 @@ function renderPost (results) {
...
-  if (fs.existsSync(filepath)) {
-    return Promise.reject(new Error(`Release post for ${results.version} already exists!`))
-  }
+  return new Promise(function (resolve, reject) {
+    fs.access(filepath, fs.F_OK, function (err) {
+      if (!err) {
+        return reject(new Error(`Release post for ${results.version} already exists!`))
+      }
+
+      fs.writeFile(filepath, results.content, function (err1) {
```
Comment (lpinca) [inline scripts/release-post.js:190]: @thefourtheye any reason for using `err1` here?
I think we can use `err` again as it will shadow the other `err`.

Comment (thefourtheye) [inline scripts/release-post.js:190]: @lpinca Exactly the point. I thought people will not be confused when they read the code with `err1`.

Comment (lpinca) [inline scripts/release-post.js:190]: Ok, I'm fine with that :)

**Comment by phillipj** [comment]
Nice work @thefourtheye, LGTM!

**Pull request merged**